engines/interp_seq_mc.cpp

```diff
@@ -172,10 +171,14 @@ bool InterpSeqMC::step(int i)
       // replay it in the solver with model generation
       concrete_cex_ = true;
     } else if (r.is_sat() && !bmc_res.is_sat()) {
-      throw PonoException("Internal error: Expecting satisfiable result");
+      if (bmc_res.is_unsat()) {
+        throw PonoException("expecting satisfiable result");
+      } else {
+        throw InternalSolverException("solver could not compute BMC query");
```

Comment:
Not sure about throwing `InternalSolverException`, since this would cross module boundaries and create tighter coupling to smt-switch. This exception class belongs to the smt-switch module. In Pono, exceptions should generally be `PonoException` or one of its derived classes.